### PR TITLE
Do not check missing ship/pay methods on turbo request

### PR DESCRIPTION
**Pull request by cyrillefr**
#### What? Why?

- Contributes to #12596

#### What should we test?

Not finalized yet, but this first version should stop displaying twice a dismiss box checking the absence of shipment/payment methods when reproducing the steps to reproduce from issue #12596.

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [X] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [ ] Technical changes only
- [X] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Comment by cyrillefr** [comment]
Hello @drummer83 ,

I am not finished (and would need help) but what I have found is here:
In admin, there is a check that is made on almost every page for absent shipment/payment method.
When on product, with a dismiss box displayed:
- on updating an element, there is a refresh of the content(not all the page), but the logic is the same: there is a check for absent shipment/payment methods.
So a second dismiss box is displayed, as well as a changes saved box.  

The code is here:
https://github.com/openfoodfoundation/openfoodnetwork/blob/a66fec0b265ccd66f6d1217de43c7f63375a0d0a/app/controllers/spree/admin/base_controller.rb#L22-L30

In an attempt to get things a little cleaner, I have restricted that check to regular requests, and not to `turbo` request (the one that refreshes only a part of the page).

So, it leaves us with a dismiss box that almost totally hides the saved changes box.
Cleaner, but it hides important information ...

Technically speaking, I have played a bit with the z-index, but there is always one box that hides the other(depending on how the z-index). Because, they are on the same location.

The 2 boxes are not in the same div, so it is not a question of how to fit the 2 boxes together in the same space. The shared div for flash messages is called once at first loading then a 2nd time at refresh. 

I am a bit clueless here ... @mkllnk , would you have an idea ?

**Review by dacook — COMMENTED** [review]
Nice investigation, thanks Cyrille.

I think this is a good first step: there's never a need for this warning to be checked multiple times. 
But I think we should rename `html_request?` to match its intention, which I think is something like `full_page_load?`. What do you think?

**Comment by dacook** [comment]
But it sounds like the different alerts are still overlapping, and it's not fixable because they are not in the same container. 
The problem is because we have [one global `#flashes` element](https://github.com/openfoodfoundation/openfoodnetwork/blob/master/app/views/spree/layouts/_admin_body.html.haml#L6-L7) that is populated on page load, but then turbo_stream responses add more `.flash-container`s in random places in the document ([example](https://github.com/openfoodfoundation/openfoodnetwork/blob/master/app/views/admin/column_preferences/bulk_update.turbo_stream.haml#L2)).

I think we need to update how we show these alerts to ensure they are always in the same container. Then we can manage with normal style rules.
We should be able to update the turbo stream layouts to do that, perhaps it's simple as this change for example:
 
[`bulk_update.turbo_stream.haml`](https://github.com/openfoodfoundation/openfoodnetwork/blob/master/app/views/admin/column_preferences/bulk_update.turbo_stream.haml)
```haml
- if defined? flash
  = turbo_stream.append "flashes" do
    = render partial: "admin/shared/flashes", locals: { flashes: flash }

= turbo_stream.replace "bulk_admin_column_preferences_form" do
  = render partial: 'form', locals: { action: }
```

File: app/controllers/spree/admin/base_controller.rb [hunk 1]
```diff
@@ -82,6 +82,9 @@ def check_json_authenticity
       private
 
       def html_request?
+        return false if request.format.symbol == :turbo_stream ||
+                        request.format.to_s.include?("turbo")
+
```
Comment (rioug) [inline app/controllers/spree/admin/base_controller.rb:87]: You'd think checking for `:turbo_stream` would be enough ? but I guess not. So I think we could just simplify it to 
```suggestion
        return false if request.format.to_s.include?("turbo")
```
What do you think ?

**Review by rioug — CHANGES_REQUESTED** [review]

**Comment by chahmedejaz** [comment]
I think we can close this one as the attached issue has been fixed and closed by: #12778 Thanks.

**Pull request closed without merging**